core/src/main/scala/cats/Divisible.scala

```diff
@@ -0,0 +1,52 @@
+package cats
+
+import simulacrum.typeclass
+
+/**
+ * Divisible functors
+ *
+ * Must obey the laws defined in cats.laws.DivisibleLaws.
+ *
+ * Based on ekmett's contravariant library:
+ * https://hackage.haskell.org/package/contravariant-1.4/docs/Data-Functor-Contravariant-Divisible.html
+ */
+@typeclass trait Divisible[F[_]] extends ContravariantSemigroupal[F] { self =>
+
+  /**
+   * `unit` produces an instance of `F` for any type `A`
+   */
+  def unit[A]: F[A]
+
+  /**
+   * `contramap2`
+   *
+   * Given two values in the Divisible context, and a function producing a value of both types,
+   * yields an element of the domain of the function lifted into the context.
+   */
+  def contramap2[A, B, C](fb: F[B], fc: F[C])(f: A => (B, C)): F[A]
```

Comment:
Why not provide a default implementation in terms of `contramap` and `product`? (like we do for covariant semigroupal: https://github.com/typelevel/cats/blob/master/core/src/main/scala/cats/Apply.scala#L44-L50 )

Reply:
Sure, that makes sense, consistency is pretty valuable. I mostly went this approach based on the primary document.